internal/http.go

```diff
@@ -50,18 +50,36 @@ func downloadReport(ctx context.Context, w http.ResponseWriter, r *http.Request,
 		return &HTTPError{HTTPErr.code, HTTPErr.message, nil}
 	}
 
-	err := generateReport(ctx, conn, r, logger)
+	query := r.URL.Query()
+	start := query["start"]
+	end := query["end"]
+	dirName := "/tmp/curator-report"
+	err := os.MkdirAll(dirName, 0755)
+	if err != nil {
+		return &HTTPError{http.StatusInternalServerError, "error creating directory", err}
+	}
+	tarDirName := fmt.Sprintf("%s/%s-%s-koku-metrics.tar.gz",
+		dirName, strings.Join(start, ","), strings.Join(end, ","))
+
+	err = generateReport(ctx, conn, r, logger, tarDirName)
 	if err != nil {
 		return &HTTPError{http.StatusInternalServerError, "error generating report", err}
 	}
 
+	f, err := os.Open(tarDirName)
+	if err != nil {
+		return &HTTPError{http.StatusInternalServerError, "error generating report", err}
+	}
+	files, err := ioutil.ReadAll(f)
+	if err != nil {
+		return &HTTPError{http.StatusInternalServerError, "error generating report", err}
+	}
 	w.WriteHeader(http.StatusOK)
-
-	_, err = w.Write([]byte("generated report"))
+	w.Header().Set("Content-Type", "application/octet-strem")
```

Comment:
Any idea on why we need this content type header? Anything wrong with `application/json` here?